### PR TITLE
Local Uicon support / On-Demand Image Generation

### PR DESCRIPTION
**Pull request by 123FLO321**
**HOW TO**
- create folder for your icon styles - use that for `git clone` multiple styles e.g. `git clone https://github.com/nileplumb/PkmnShuffleMap` - embed that parent folder as volume in docker-compose
- or use your existing images volume of Docker for that
- you can also embed each icon repo as volume separately, depends on you
- be careful, RDM needs write access in the repo folder - it will generate few icons

**ICON STYLES**
- set styles in Dashboard settings
- format `name;path` -- be careful, folder has to exist within Project-Dir `/resources/webroot/static/img/`
- user can choose from that styles in UI
- see screenshots

**UICON repo examples**
- https://github.com/nileplumb/PkmnShuffleMap/tree/master/UICONS
- https://github.com/nileplumb/PkmnHomeIcons/tree/master/UICONS
- https://github.com/jms412/PkmnShuffleMap
- https://github.com/jms412/PkmnHomeIcons
- https://github.com/whitewillem/PogoAssets/tree/main/uicons
- https://github.com/whitewillem/PogoAssets/tree/main/uicons-outline

## Screenshots
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Fabio1988** [comment]
UICON repos are missing following files:

- navigation
- some files in misc, e.g. sparkle, found, missing (used for stats)

File: resources/webroot/static/js/i18n.min.js [hunk 1]
```diff
@@ -1 +1,198 @@
...OriginalText(i||e,t,n)}return o};e.prototype.findTranslation=function(e,t,n,r){var i,s,o,u,a;o=r[e];if(o==null){return null}if(t==null){if(typeof o==="string"){return this.applyFormatting(o,t,n)}}else{if(o instanceof Array||o.length){for(u=0,a=o.length;u<a;u++){s=o[u];if((t>=s[0]||s[0]===null)&&(t<=s[1]||s[1]===null)){i=this.applyFormatting(s[2].replace("-%n",String(-t)),t,n);return this.applyFormatting(i.replace("%n",String(t)),t,n)}}}}return null};e.prototype.getContextData=function(e,t){var n,r,i,s,o,u,a,f;if(e.contexts==null){return null}a=e.contexts;for(o=0,u=a.length;o<u;o++){n=a[o];r=true;f=n.matches;for(i in f){s=f[i];r=r&&s===t[i]}if(r){return n}}return null};e.prototype.useOriginalText=function(e,t,n){if(t==null){return this.applyFormatting(e,t,n)}return this.applyFormatting(e.replace("%n",String(t)),t,n)};e.prototype.applyFormatting=function(e,t,n){var r,i;for(r in n){i=new RegExp("%{"+r+"}","g");e=e.replace(i,n[r])}return e};return e}();n=new e;t=n.translate;t.translator=n;t.create=function(n){var r;r=new e;if(n!=null){r.add(n)}r.translate.create=t.create;return r.translate};(typeof module!=="undefined"&&module!==null?module.exports=t:void 0)||(this.i18n=t)}).call(this)
```
Comment (123FLO321) [inline resources/webroot/static/js/i18n.min.js:1]: why did this change?

Comment (Fabio1988) [reply resources/webroot/static/js/i18n.min.js:1]: It's only styling (formatting), nothing changed in there :)

**Review by Fabio1988 — APPROVED** [review]

**Review by Fabio1988 — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -15,6 +15,7 @@ RealDeviceMap.xcodeproj/
 .idea
 
 resources/webroot/static/img/
+resources/webroot/static/data/www_de.json
```
Comment (123FLO321) [inline .gitignore:18]: maybe ignore all www_*.json files in that folder?

File: resources/webroot/index.js.mustache [hunk 3]
```diff
@@ -78,7 +78,7 @@ let availableItems = []
 let availableTileservers = {}
 let selectedTileserver = 'Default'
 let availableIconStyles = {}
-let selectedIconStyle = 'Default'
+let selectedIconStyle
```
Comment (123FLO321) [inline resources/webroot/index.js.mustache:81]: Shouldn't this be
```let selectedIconStyle = '{{default_icon_style}}'```
?

Comment (Fabio1988) [reply resources/webroot/index.js.mustache:81]: It's set during `function loadStorage()` - availableIconStyles does contain the default one too, why 2 params? :)

Comment (Fabio1988) [reply resources/webroot/index.js.mustache:81]: fixed that in latest commit

**Pull request merged**